### PR TITLE
setup: upgrade edtf to 5.0.0 and fix bug with dates and times with hour 23 & release: v1.2.0

### LINKED ISSUE #11: Upgrade to edtf 5 to fix bug with date time and hour 23
Package version (if known): v1.1.4

## Describe the bug

Parsing a date and time at the hour 23 fails parsing (e.g. "2004-01-01T23:20:30-04:00").

## Steps to Reproduce

1. Create a new draft record
2. In the "Recommended information" section, under "Dates", enter `1985-04-12T22:20:30` in the "Date" field, and save
--> :white_check_mark: The field appears as valid
3. Enter `1985-04-12T23:20:30` in the same "Date" field, and save
--> :x: The field appears as invalid

## Expected behavior

Entering at time at hour 23 should work.

## Additional context

The version 4.0.1 of the PyPI package `edtf` (repo [`ixc/python-edtf`](https://github.com/ixc/python-edtf)) which we are currently using is the source of this bug.
 has the same bug, while version 5.0.0 does not have this bug.
The version 5.0.0 was released in July 2024 after many years of inactivity, and itdoes not have this bug.

## Screenshot

Error shown when entering and date and time with the hour 23

### PR DESCRIPTION
**Pull request by ptamarit**
Fixes #11

:heart: Thank you for your contribution!

### Description

See description of the bug in #11.

* The bug is coming from the underlying version 4.0.1 of the PyPI package `edtf` (repo [`ixc/python-edtf`](https://github.com/ixc/python-edtf)) which we are currently using.
* The version 5.0.0 of this package was released in July 2024 after many years of inactivity.
* Version 5.0.0 does not have this bug.

The pull request can be reviewed commit by commit.

### Checklist

Ticks in all boxes and 🟢 on all GitHub actions status checks are required to merge:

- [x] I'm aware of the [code of conduct](https://inveniordm.docs.cern.ch/contribute/code-of-conduct/).
- [x] I've created [logical separate commits](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commits) and followed the [commit message format](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#commit-message).
- [x] I've added relevant test cases.
- [x] I've added relevant documentation.
- [x] I've marked [translation strings](https://inveniordm.docs.cern.ch/develop/howtos/i18n/).
- [x] I've identified the [copyright holder(s)](https://inveniordm.docs.cern.ch/contribute/copyright-policy/) and updated copyright headers for touched files (>15 lines contributions).
- [x] I've NOT included third-party code (copy/pasted source code or new dependencies).
    * If you have added [third-party code (copy/pasted or new dependencies)](https://inveniordm.docs.cern.ch/develop/best-practices/commits/#third-party-codedependencies), please reach out to an [architect](https://github.com/orgs/inveniosoftware/teams/architects/members).

**Frontend**

- [x] I've followed the [CSS/JS](https://inveniordm.docs.cern.ch/develop/best-practices/css-js/) and [React](https://inveniordm.docs.cern.ch/develop/best-practices/react/) guidelines.
- [x] I've followed the [web accessibility](https://inveniordm.docs.cern.ch/develop/best-practices/accessibility/) guidelines.
- [x] I've followed the [user interface](https://inveniordm.docs.cern.ch/develop/best-practices/ui/) guidelines.

**Reminder**

By using GitHub, you have already agreed to the [GitHub’s Terms of Service](https://help.github.com/articles/github-terms-of-service/#6-contributions-under-repository-license) including that:

1. You license your contribution under the same terms as the current repository’s license.
2. You agree that you have the right to license your contribution under the current repository’s license.

### CODE REVIEW

File: tests/test_api.py [hunk 1]
```diff
@@ -162,10 +169,31 @@ def test_format_edtf_format():
...
+    ),
+    # Date and time with timeshift in hours (only)
+    (
+        '1985-04-12T23:20:30-04',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30-04')
+    ),
+    # Date and time with timeshift in hours and minutes
+    (
+        '1985-04-12T23:20:30+04:30',
+        DateAndTime(Date('1985', '04', '12'), '23:20:30+04:30')
```
Comment (ptamarit) [inline tests/test_api.py:191]: Running these tests with version `v4.0.1` of `edtf` was failing with the following, while upgrading to `v5.0.0` fixes the bug.

```
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
babel_edtf/__init__.py:213: in parse_edtf
    edtf_date = edtf_parse_edtf(date)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

str = '2004-01-01T23:20:30-04:00', parseAll = True, fail_silently = False

    def parse_edtf(str, parseAll=True, fail_silently=False):
        try:
            if not str:
                raise ParseException("You must supply some input text")
            p = edtfParser.parseString(str.strip(), parseAll)
            if p:
                return p[0]
        except ParseException as e:
            if fail_silently:
                return None
>           raise EDTFParseException(e)
E           edtf.parser.edtf_exceptions.EDTFParseException: Expected end of text, found 'T23'  (at char 10), (line:1, col:11)  (at char 0), (line:1, col:1)

.venv/lib/python3.9/site-packages/edtf/parser/grammar.py:291: EDTFParseException

========================================================================================================== short test summary info ===========================================================================================================
FAILED tests/test_api.py::test_parse_edtf[2004-01-01T23:20:30-04:00-expected8] - edtf.parser.edtf_exceptions.EDTFParseException: Expected end of text, found 'T23'  (at char 10), (line:1, col:11)  (at char 0), (line:1, col:1)
================================================================================================== 1 failed, 76 passed, 15 skipped in 1.56s ==================================================================================================
```

File: babel_edtf/__init__.py [hunk 2]
```diff
@@ -13,11 +13,12 @@
...
 from babel.dates import LC_TIME, format_date, format_interval, format_skeleton
 from dateutil.parser import isoparse
-from edtf import PRECISION_DAY, PRECISION_MONTH, PRECISION_YEAR, Date, \
-    EDTFObject, Interval
+from edtf import Date, EDTFObject, Interval
 from edtf import parse_edtf as edtf_parse_edtf
 from edtf import struct_time_to_datetime
 from edtf.parser.grammar import EDTFParseException, ParseException
+from edtf.parser.parser_classes import PRECISION_DAY, PRECISION_MONTH, \
+    PRECISION_YEAR
```
Comment (ptamarit) [inline babel_edtf/__init__.py:21]: These `PRECISION_` constants are not re-exported in `edtf` anymore.

**Review by jrcastro2 — APPROVED** [review]
LGTM!

**Comment by ptamarit** [comment]
Just updated the release date in the changelog.

**Comment by mfenner** [comment]
@ptamarit thanks for fixing this! It would be good to also update downstream dependencies such as `marshmallow-utils` and `invenio-records-resources` so that this fix can also be applied to current InvenioRDM instances (v12 and maybe earlier versions).

**Comment by ptamarit** [comment]
One thing we noticed is that the previous version of the library could parse an input like `"2024 - 11 - 07"` while the new version does not accept spaces.

**Comment by ptamarit** [comment]
> @ptamarit thanks for fixing this! It would be good to also update downstream dependencies such as `marshmallow-utils` and `invenio-records-resources` so that this fix can also be applied to current InvenioRDM instances (v12 and maybe earlier versions).

I did [bump dependencies on the InvenioRDM v12 demo instance](https://github.com/inveniosoftware/demo-inveniordm/commit/5ca4579f033aa0a89525b09f3dcc1bde305dcea9) and then managed to publish a record with the date `1985-04-12T23:20:30`.

**Comment by ptamarit** [comment]
> thanks for fixing this! It would be good to also update downstream dependencies such as `marshmallow-utils` and `invenio-records-resources` so that this fix can also be applied to current InvenioRDM instances (v12 and maybe earlier versions).

@mfenner I did [bump dependencies on the InvenioRDM v12 demo instance](https://github.com/inveniosoftware/demo-inveniordm/commit/5ca4579f033aa0a89525b09f3dcc1bde305dcea9) and then managed to publish a record with the date `1985-04-12T23:20:30` on the [InvenioRDM demo instance](https://inveniordm.web.cern.ch/records/1h2qb-fp597).

**Pull request merged**